### PR TITLE
feat: Add scrollIntoView method to Component

### PR DESCRIPTION
**Pull request by mstahv**
## Description

This change adds back a scrollIntoView method that we had in past framework versions.

The implementation is a trivial shorthand to the similarly named DOM method, and very handy in UI development.
It has been requested by multiple SoF and forum users, but apparently nobody wants to write this trivial enhancement issues as I couldn't find an existing issue for this.

The implementation leaves door open for overloading the method with a version with additional parameters, but already this method should solve most use cases.

## Type of change

- [ ] Bugfix
- [x] Feature

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/component/Component.java [hunk 1]
```diff
@@ -676,4 +676,12 @@ protected Locale getLocale() {
         }
         return locale;
     }
+
+    /**
+     * Scrolls the current component into the visible area of the browser
+     * window.
+     */
+    public void scrollIntoView() {
+        getElement().callJsFunction("scrollIntoView");
```
Comment (joheriks) [inline flow-server/src/main/java/com/vaadin/flow/component/Component.java:685]: 1. Would it make sense to add `scrollIntoView` also to the `com.vaadin.flow.dom.Element` API and delegate to that?

2. What is the precondition for calling this method? When the component is not attached, e.g. in a constructor, the JS invocation is postponed until the component is attached but it does not scroll still. It also does not work when called from an attachlistener. It could either be documented that the caller must ensure the page is fully rendered before calling this method, or the scrolling action should be deferred somehow.

**Review by joheriks — CHANGES_REQUESTED** [review]

**Comment by joheriks** [comment]
Should be rebased.

File: flow-server/src/main/java/com/vaadin/flow/component/Component.java (hunk 1 above)
Comment (mstahv) [reply flow-server/src/main/java/com/vaadin/flow/component/Component.java]: Don't really know why it didn't work on new instances, but setTimeout hack fixed it. Moved the implementation to Element.

**Comment by caalador** [comment]
I guess something went wrong with the rebase `+3,451 −2,208 `

**Comment by mstahv** [comment]
I guess so, and now it shows conflicts 🤷‍♂️

File: flow-tests/test-root-context/src/main/java/com/vaadin/flow/uitest/ui/scroll/ScrollableView.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2000-2021 Vaadin Ltd.
```
Comment (joheriks) [inline flow-tests/test-root-context/src/main/java/com/vaadin/flow/uitest/ui/scroll/ScrollableView.java:2]:
```suggestion
 * Copyright 2000-2022 Vaadin Ltd.
```

File: flow-server/src/main/java/com/vaadin/flow/dom/Element.java [hunk 3]
```diff
@@ -1603,4 +1603,19 @@ private void informChildrenOfStateChange(boolean enabled, Element element) {
...
+     * @see <a href=
+     *      "https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView">Mozilla
+     *      docs</a>
+     * @return the element
+     */
+    public Element scrollIntoView() {
+        // for an unknown reason, needs to be called deferred to work on a newly
+        // created element
+        executeJs(
+                "var el = this; setTimeout(function() {el.scrollIntoView();}, 0);");
```
Comment (joheriks) [inline flow-server/src/main/java/com/vaadin/flow/dom/Element.java:1618]: I guess this works because the JS is deferred long enough until the page rendering has stabilized. Looks fine to me, but should be tested on the major browsers. Which have you tested so far?  I will test the rest.

Comment (mstahv) [reply flow-server/src/main/java/com/vaadin/flow/dom/Element.java:1618]: Safari, Chrome, FF on Mac.

Comment (joheriks) [reply flow-server/src/main/java/com/vaadin/flow/dom/Element.java:1618]: Also tested Windows Edge and iOS Safari, seemed to scroll fine.

**Review by joheriks — APPROVED** [review]
